tests/log_fidelity/test_check_syslog.py

```diff
@@ -0,0 +1,45 @@
+import logging
+import pytest
+
+from tests.common.plugins.loganalyzer.loganalyzer import LogAnalyzer, LogAnalyzerError
+
+logger=logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any')
+]
+
+@pytest.fixture
+def trigger_action():
+    BGP_DOWN_COMMAND = "sudo config bgp shutdown all"
+    return BGP_DOWN_COMMAND
+
+@pytest.fixture
+def bgp_shutdown_log():
+    BGP_DOWN_EXPECTED_LOG_MESSAGE = "admin state is set to 'down'"
+    return BGP_DOWN_EXPECTED_LOG_MESSAGE
+
+@pytest.fixture
+def restore_action():
+    BGP_UP_COMMAND = "sudo config bgp startup all"
+    return BGP_UP_COMMAND
+
+def test_check_syslog(duthosts, rand_one_dut_hostname, bgp_shutdown_log, trigger_action, restore_action):
+    duthost=duthosts[rand_one_dut_hostname]
+    loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix="bgp_shutdown")
+    loganalyzer.expect_regex=[]
+
+    try:
+        loganalyzer.expect_regex.append(bgp_shutdown_log)
+        marker=loganalyzer.init()
+        duthost.command(trigger_action)
+        logger.info("Check for expected log in syslog")
+        loganalyzer.analyze(marker)
+
+    except LogAnalyzerError as err:
+        logger.error("Unable to find expected log in syslog")
+        raise err
+
+    finally:
+        duthost.command(restore_action)
```

Comment:
I think what @yxieca was suggesting was that you take this code and stick it into a helper method (like "check_syslog" or something similar) that takes in a marker prefix, expected command, expected regex, and a recovery action. With that helper method you can re-write the test as:

```
def test_bgp_shutdown_syslog(duthosts, rand_one_dut_hostname):
    check_syslog("bgp_shutdown", BGP_DOWN_COMMAND, BGP_DOWN_EXPECTED_MESSAGE, BGP_UP_COMMAND)
```

And then later if you/me/someone else adds a new scenario and wants to write a test for it it will only take ~2ish lines of code to do so. :smile:

Alternatively, pytest also has this [parameterization feature](https://docs.pytest.org/en/stable/parametrize.html) that you could try to use (I don't know if that saves us much effort compared to the first approach, but it is a handy trick to know about).

Reply:
I see! Thanks for the clarification 😄